src/filters.jl

```diff
@@ -152,12 +152,16 @@ function sample_proposal_and_compute_log_weights!(
     rng::Random.AbstractRNG,
 )
     num_particle = size(states, 2)
-    Threads.@threads :static for p in 1:num_particle
-        update_state_deterministic!(selectdim(states, 2, p), model, time_index)
-        update_state_stochastic!(selectdim(states, 2, p), model, rng)
-        log_weights[p] = get_log_density_observation_given_state(
-            observation, selectdim(states, 2, p), model
-        )
+    nchunks = cld(num_particle, Threads.nthreads())
```

Comment:
I think `nchunks` here is the size of each chunk rather than the number of chunks the particle indices are split in to as [`Iterators.partition(collection, n)` partitions `collection` in to sub-collections of size `n`](https://docs.julialang.org/en/v1/base/iterators/#Base.Iterators.partition).